hedera-node/src/main/java/com/hedera/services/txns/token/TokenFreezeUnfreezeLogic.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2020-2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.txns.token;
+
+import com.hedera.services.store.models.Id;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+public interface TokenFreezeUnfreezeLogic {
+    ResponseCodeEnum validate(TransactionBody txnBody);
+
+    void doFreezeUnfreeze(Id targetTokenId, Id targetAccountId);
+}
```

Comment:
The design would be cleaner if you remove the interface and keep 2 separate `Logic` implementations -> `FreezeLogic`, `UnfreezeLogic` without having them extend a common class/interface.

Having `FreezeLogic` with `doFreezeUnfreeze` method may imply that there is an option to do `unfreeze` operation inside a `FreezeLogic` and vice versa. This is confusing.